doppler/resource_service_token.go

```diff
@@ -98,7 +99,8 @@ func resourceServiceTokenRead(ctx context.Context, d *schema.ResourceData, m int
 	}
 
 	if token == nil {
-		return diag.Errorf("Could not find service token")
+		err := errors.New("Doppler Error: Could not find requested service token")
+		return handleNotFoundError(err, d)
```

Comment:
Our custom error will no longer register with `handleNotFoundError` because it's not an `APIError`. One solution could be to introduce a custom error. I haven't tested this but it could look something like:

```diff
diff --git a/doppler/provider_utils.go b/doppler/provider_utils.go
index 8aa6123..26fc5b7 100644
--- a/doppler/provider_utils.go
+++ b/doppler/provider_utils.go
@@ -7,8 +7,26 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+type CustomNotFoundError struct {
+	Message string
+}
+
+func (e *CustomNotFoundError) Error() string {
+	return fmt.Sprintf("Doppler Error: %s", e.Message)
+}
+
 func handleNotFoundError(err error, d *schema.ResourceData) diag.Diagnostics {
+	isNotFoundError := false
+
 	if apiError, ok := err.(*APIError); ok && apiError.Response != nil && apiError.Response.HTTPResponse.StatusCode == 404 {
+		isNotFoundError = true
+	}
+
+	if _, ok := err.(*CustomNotFoundError); ok {
+		isNotFoundError = true
+	}
+
+	if isNotFoundError {
 		// the resource no longer exists, so reset its ID so Terraform will
 		// generate a plan that recreates it
 		d.SetId("")
@@ -16,7 +34,7 @@ func handleNotFoundError(err error, d *schema.ResourceData) diag.Diagnostics {
 			diag.Diagnostic{
 				Severity: diag.Warning,
 				Summary:  err.Error(),
-				Detail:   fmt.Sprintf("Resource was not found, so it was removed from state and is being recreated. Status: %v", apiError.Response.HTTPResponse.StatusCode),
+				Detail:   "Resource was not found, so it was removed from state and is being recreated.",
 			},
 		}
 	}
diff --git a/doppler/resource_service_token.go b/doppler/resource_service_token.go
index 4b8ebf1..5f17485 100644
--- a/doppler/resource_service_token.go
+++ b/doppler/resource_service_token.go
@@ -2,7 +2,6 @@ package doppler
 
 import (
 	"context"
-	"errors"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -99,7 +98,7 @@ func resourceServiceTokenRead(ctx context.Context, d *schema.ResourceData, m int
 	}
 
 	if token == nil {
-		err := errors.New("Doppler Error: Could not find requested service token")
+		err := &CustomNotFoundError{Message: "Could not find requested service token"}
 		return handleNotFoundError(err, d)
 	}
 


```

Reply:
Made this change and seems to be working well!